pkg/trustedresources/verify.go

```diff
@@ -88,7 +105,7 @@ func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.
 }
 
 // VerifyPipeline verifies the signature and public key against pipeline
-func VerifyPipeline(ctx context.Context, pipelineObj v1beta1.PipelineObject, k8s kubernetes.Interface) error {
+func VerifyPipeline(ctx context.Context, pipelineObj v1beta1.PipelineObject, k8s kubernetes.Interface, namespace, source string, verificationpolicies []*v1alpha1.VerificationPolicy) error {
```

Comment:
We should document what `source` is and what the expected value(s) should be. We're passing in `""` as the value for a lot of places which may be a sign this isn't the best interface for this func.

Reply:
Similar for verification policies - having to plumb through a variable across a bunch of different methods to use it somewhere further down the stack is also a sign that this may not be the right abstraction.

i.e. it might be a good thought experiment if you had a minimal interface "PipelineVerifier" interface like 

```go
type PipelineVerifier interface{
  func VerifyPipeline(ctx context.Context, obj v1beta1.PipelineObject) error
}
```

and see what changes would need to be made to configure things 🤔 

(I suspect this will probably be a larger refactor for how resources are fetched during reconcile, so this would be worth doing in another PR).

Reply:
For local resolver we don't have a `source` yet,  that's why we pass "". 
yes this is a great suggestion! I think we can refactor this. I will mark this as a todo if it needs too many changes

Reply:
I added how we use configsource and patterns in the doc, and also update the function docs.

Reply:
I'm still not a huge fan of plumbing the verification policies all the way through. 😕

As a compromise - how about creating a new func provider that wraps the existing `resources.GetPipelineFunc`? e.g.

```go
func GetVerifiedPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clientset.Interface, requester remoteresource.Requester, pipelineRun *v1beta1.PipelineRun, verificationpolicies []*v1alpha1.VerificationPolicy) (rprp.GetPipeline, error) {
	get, err := resources.GetPipelineFunc(ctx, c.KubeClientSet, c.PipelineClientSet, c.resolutionRequester, pr)

	return func(context.Context, string) (v1beta1.PipelineObject, *v1beta1.ConfigSource, error) {
		p, err := get(...)
		if err != nil {
			...
		}
		if err := VerifyPipeline(...); err != nil {
			return nil, nil, err
		}
		return p, nil
	}, nil
}
```

This way fetching is still separate from verification, we don't need to plumb the policies all the way through, and we can easily refactor this into an interface later.

Reply:
I have refactored this part, thanks a lot! The only diff is that I cannot use rprp.GetPipeline as a return directly. It's getting syntax error